loadgen/cmd/otelbench/remote_stats.go

```diff
@@ -0,0 +1,236 @@
+	},
+	"otelcol_receiver_refused_log_records": {
+		Sum: &types.SumAggregation{Field: some.String("otelcol_receiver_refused_log_records")},
+	},
+	"otelcol_receiver_refused_metric_points": {
+		Sum: &types.SumAggregation{Field: some.String("otelcol_receiver_refused_metric_points")},
+	},
+	"otelcol_receiver_refused_spans": {
+		Sum: &types.SumAggregation{Field: some.String("otelcol_receiver_refused_spans")},
+	},
+	"otelcol_scraper_errored_metric_points": {
+		Sum: &types.SumAggregation{Field: some.String("otelcol_scraper_errored_metric_points")},
+	},
+	"otelcol_scraper_scraped_metric_points": {
+		Sum: &types.SumAggregation{Field: some.String("otelcol_scraper_scraped_metric_points")},
+	},
+}
+
+type elasticsearchTelemetryConfig TelemetryConfig
+
+func (cfg *elasticsearchTelemetryConfig) validate() error {
+	if len(cfg.ElasticsearchURL) == 0 {
+		return errors.New("remote stats will be ignored because of empty telemetry Elasticsearch URL")
+	}
+	if cfg.ElasticsearchAPIKey == "" && (cfg.ElasticsearchUserName == "" || cfg.ElasticsearchPassword == "") {
+		return errors.New("remote stats will be ignored because of empty telemetry Elasticsearch API key and username or password")
+	}
+	if cfg.ElasticsearchIndex == "" {
+		return errors.New("remote stats will be ignored because of empty telemetry Elasticsearch search index")
+	}
+	// Use the default list of metrics if empty list was provided.
+	if len(cfg.Metrics) == 0 {
+		cfg.Metrics = []string{
+			"otelcol_process_cpu_seconds",
+			"otelcol_process_memory_rss",
+			"otelcol_process_runtime_total_alloc_bytes",
+			"otelcol_process_runtime_total_sys_memory_bytes",
+			"otelcol_process_uptime",
+		}
+	}
+	return nil
+}
+
+type remoteStatsFetcher interface {
+	FetchStats(ctx context.Context, from, to time.Time) (map[string]float64, error)
+}
+
+type elasticsearchStatsFetcher struct {
+	elasticsearchTelemetryConfig
+	client *elasticsearch.TypedClient
+}
+
+func newElasticsearchStatsFetcher(cfg elasticsearchTelemetryConfig) (remoteStatsFetcher, error) {
+	if err := cfg.validate(); err != nil {
+		return nil, err
+	}
+	client, err := elasticsearch.NewTypedClient(elasticsearch.Config{
+		Addresses: cfg.ElasticsearchURL,
+		Username:  cfg.ElasticsearchUserName,
+		Password:  cfg.ElasticsearchPassword,
+		APIKey:    cfg.ElasticsearchAPIKey,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return elasticsearchStatsFetcher{elasticsearchTelemetryConfig: cfg, client: client}, nil
+}
+
+func (esf elasticsearchStatsFetcher) FetchStats(ctx context.Context, from, to time.Time) (map[string]float64, error) {
+	var filters []types.Query
+	if esf.FilterCluster != "" {
+		filters = append(filters, types.Query{
+			Term: map[string]types.TermQuery{
+				"labels.cluster": {
+					Value: esf.FilterCluster,
+				},
+			},
+		})
+	}
+	if esf.FilterProject != "" {
+		filters = append(filters, types.Query{
+			Term: map[string]types.TermQuery{
+				"labels.project": {
```

Comment:
```suggestion
				"labels.project_id": {
```

Q: shouldn't it be `labels.project_id`?

Reply:
updated